pkg/collector/stats/types/types.go

```diff
@@ -132,8 +132,16 @@ func (s *UInt64StatCollection) SetDeltaStat(key string, newDelta uint64) {
 // SumAllDeltaValues aggregates the delta metrics of all sources (i.e., stat keys)
 func (s *UInt64StatCollection) SumAllDeltaValues() uint64 {
 	sum := uint64(0)
+	if s == nil {
+		klog.V(3).Info(" s  == nil")
+		return sum
+	}
 	for _, stat := range s.Stat {
-		sum += stat.Delta
+		if stat != nil {
+			sum += stat.Delta
+		} else {
+			klog.V(3).Info(" Error retrieving stat")
+		}
 	}
 	return sum
```

Comment:
This seems to be fixing an issue than a required patch for the abstraction. Is that right? 
Also makes me wonder when is it that `s` and `stat` can be nil?

Reply:
I can't remember what HW and test generated a null pointer dereference but this is something that manifested in one of the setups when I ran kepler

Reply:
IMO, calling `SumAllDeltaValues()` on a nil UInt64StatCollection   is a logical error and should be fixed than masking it.